docker/Dockerfile

```diff
@@ -62,6 +62,21 @@ RUN apt-get update && \
     rm -rf /var/lib/apt/lists/*
 RUN /usr/local/bin/recursive_vcs_import.py src src/upstream
 
+# remove blacklisted packages from workspace
+ARG BLACKLISTED_PACKAGES_FILE="docker/blacklisted-packages.txt"
+ARG ENABLE_RECURSIVE_BLACKLISTED_PACKAGES="false"
+RUN echo "colcon list -p --base-paths src/ --packages-select \\" >> $WORKSPACE/.remove-packages.sh && \
+    if [[ $ENABLE_RECURSIVE_BLACKLISTED_PACKAGES == 'true' ]]; then \
+        find . -type f -name $(basename ${BLACKLISTED_PACKAGES_FILE}) -exec sed '$a\' {} \; | awk '{print "  " $0 " \\"}' >> $WORKSPACE/.remove-packages.sh ; \
+    elif [[ -f src/target/${BLACKLISTED_PACKAGES_FILE} ]]; then \
+        cat src/target/${BLACKLISTED_PACKAGES_FILE} | awk '{print "  " $0 " \\"}' >> $WORKSPACE/.remove-packages.sh ; \
+    fi && \
+    echo ";" >> $WORKSPACE/.remove-packages.sh && \
+    chmod +x $WORKSPACE/.remove-packages.sh && \
+    $WORKSPACE/.remove-packages.sh | xargs rm -rf
+
+# TODO: only keep whitelist packages in workspace
+
```

Comment:
I suggest that we still implement this, even though it might be cumbersome to explicitly whitelist *all* packages that should be kept. A more complex feature would be to allow glob-patterns in those files, but I would keep that for the future.

Whitelisting should have precedence over blacklisting.

I also suggest to assume that whitelisting/blacklisting only considers upstream packages anyway.

Reply:
Sounds reasonable to me. We should implement it that way.

Reply:
Skipped for now.